src/library_eventloop.js

```diff
@@ -122,16 +122,15 @@ LibraryJSEventLoop = {
   emscripten_set_timeout_loop__deps: ['$callUserCallback'],
   emscripten_set_timeout_loop: function(cb, msecs, userData) {
     function tick() {
-      var t = performance.now();
-      var n = t + msecs;
-      {{{ runtimeKeepalivePop() }}}
+      // Schedule the new setTimeout() immediately when this cb fires, so that the next cb suffers
+      // as little as possible from the mandated minimum 4ms setTimeout throttling delay.
+      // If the user does not want to continue the loop, we will cancel it below.
+      var t = performance.now(),
+        ev = setTimeout(msecs);
```

Comment:
```suggestion
        ev = setInterval(msecs);
```
(unless I've misunderstood something?)

Reply:
The idea with `emscripten_set_timeout_loop()` is to explicitly/manually use `setTimeout()` to implement a loop, rather than using `setInterval()`. The thinking here was that people used to complain about subtle differencies between the heuristic scheduling logic between setTimeout() and setInterval() functions in the past, that both APIs were provided, letting the user choose which one to utilize. That is why there are `emscripten_set_timeout_loop()` and `emscripten_set_interval()` separately.